docs/high/dataset.rst

```diff
@@ -303,18 +314,6 @@ list of points to select, so be careful when using it with large masks::
    Selecting using an empty list is now allowed.
    This returns an array with length 0 in the relevant dimension.
 
-.. _dataset_iter:
```

Comment:
Can you preserve this line above the header? This makes a label for cross-linking.

To make internal links to a label, you'd write:

```rst
:ref:`dataset_iter` (using the text from the heading)
:ref:`iterating over datasets <dataset_iter>` (overriding it with custom text)
```

Reply:
Ok I can try to preserve that, but it gave an error in the build of docs. I changed the line below it from a section to a subsection and there were no other references to it. How would you add that line and avoid the error? I think the error said that it required a caption (?) if it wasn't followed by a section header.

Reply:
It should be OK if it's just above the header, like the other ones. The heading level (underline style) shouldn't matter.

You can try building the docs locally with `make html` if you have a local checkout. Or I can try to have a look at some point.